consul/src/main/java/com/linecorp/armeria/internal/consul/HealthClient.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal.consul;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.QueryParams;
+
+/**
+ * Health client supports below APIs.
+ * {@code HealthClient} is responsible for endpoint of Consul API:
+ * {@code `/health`}(https://www.consul.io/api/health.html)
+ *
+ * <p>GET /health/node/:node
+ * GET /health/checks/:service
+ * GET /health/service/:service?passing
+ * GET /health/connect/:service
+ * GET /health/state/:state
+ */
+final class HealthClient {
+
+    private static final Logger logger = LoggerFactory.getLogger(HealthClient.class);
+    private final ConsulClient client;
+    private final ObjectMapper mapper;
+
+    public static HealthClient of(ConsulClient consulClient) {
+        return new HealthClient(consulClient);
+    }
+
+    private HealthClient(ConsulClient client) {
+        this.client = client;
+        mapper = client.getObjectMapper();
+    }
+
+    /**
+     * Gets a list of nodes for service.
+     */
+    private CompletableFuture<List<HealthService>> service(String serviceName, QueryParams params) {
+        logger.trace("service() {}, {}", serviceName, params.toQueryString());
+        return client.consulWebClient()
+                     .get("/health/service/" + serviceName + '?' + params.toQueryString())
+                     .aggregate()
+                     .thenApply(response -> {
+                         final List<HealthService> healths = new ArrayList<>();
+                         logger.trace("response: {}", response);
+                         if (!Objects.equals(response.status(), HttpStatus.OK)) {
```

Comment:
Could we leave `WARN` message if the response status is not 200 OK.
https://github.com/line/armeria/blob/c7954d9b49871ebdb66bf055caa9ba40f3662f91/eureka/src/main/java/com/linecorp/armeria/client/eureka/EurekaEndpointGroup.java#L202-L206

Reply:
If we receive an unexpected status from Consul, please don't update the endpoints with an empty set.
It makes a system high available from outages.

Reply:
Oh, I agree about that.
I fixed that like this:
```
                         final HttpStatus status = response.status();
                         if (!status.isSuccess()) {
                             logger.warn("Unexpected response from: {}. (status: {}, content: {})",
                                         client.consulWebClient().uri(), status, response.contentUtf8());
                             throw new CompletionException(
                                     new RuntimeException("Unexpected response. status: " + status));
                         }
                         try {
                             return ImmutableList.copyOf(mapper.readValue(
                                     response.content().toStringUtf8(), HealthService[].class));
                         } catch (IOException e) {
                             throw new CompletionException(e);
                         }
```
And please take a look `ConsulEndpointGroup` also.